### PR TITLE
Add Filter by/out on TDS Result to post filter 

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary
Fixes https://github.com/finos/legend-studio/issues/1204

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/73408381/177859454-de9ee8ea-d6ac-40f5-8f4a-3870e9ca9c70.mov

### CODE REVIEW

**Comment by YannanGao-gs** [comment]
- [ ] Confirmed with John Pentelow, @MauricioUyaguari, and @akphi, we all agree to add a post-filter for the filter by/out operation.
- [ ] Range Selection on cell won't be supported since it is enterprise-level. https://www.ag-grid.com/react-data-grid/range-selection/

File: packages/legend-query/src/components/QueryBuilderResultPanel.tsx [hunk 1]
```diff
@@ -39,17 +42,244 @@ import {
...
+    const theProjectionColumnState =
+      queryBuilderState.fetchStructureState.projectionState.columns
+        .filter((c) => c.columnName === event?.column.getColId())
+        .concat(
+          queryBuilderState.fetchStructureState.projectionState.aggregationState.columns
+            .filter((c) => c.columnName === event?.column.getColId())
+            .map((ag) => ag.projectionColumnState),
+        )[0];
+
+    const getExistingPostFilterNode = (
```
Comment (MauricioUyaguari) [inline packages/legend-query/src/components/QueryBuilderResultPanel.tsx:99]: Your method return nodes not a singular node. We can either rename the method or return a singular node

File: packages/legend-query/src/components/QueryBuilderResultPanel.tsx [hunk 2]
```diff
@@ -39,17 +42,244 @@ import {
...
+            isFilterBy ? postFilterInOperator : postFilterNotInOperator,
+          );
+          (conditionState.value as InstanceValue).values = [
+            currentValueSpecificaton,
+            newValueSpecification,
+          ];
+        }
+      } else {
+        if (
+          !(
```
Comment (MauricioUyaguari) [inline packages/legend-query/src/components/QueryBuilderResultPanel.tsx:204]: when do you encounter this case ?

Comment (YannanGao-gs) [reply packages/legend-query/src/components/QueryBuilderResultPanel.tsx:237]: existing post filter: id is 1, user action: filter by id = 1 
existing post filter:  id is in [1 ,2 ], user action : filter by id = 1

Comment (MauricioUyaguari) [reply packages/legend-query/src/components/QueryBuilderResultPanel.tsx:231]: as discussed, lets cleanup the if statements to make it easier to read.

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**